homeassistant/components/fritzbox/config_flow.py

```diff
@@ -0,0 +1,143 @@
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+# pylint:disable=unused-import
+from .const import DOMAIN
+
+DEFAULT_HOST = "fritz.box"
+DATA_SCHEMA_USER = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+DATA_SCHEMA_CONFIRM = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+RESULT_AUTH_FAILED = "auth_failed"
+RESULT_NOT_FOUND = "not_found"
+RESULT_SUCCESS = "success"
+
+
+class FritzboxConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a AVM Fritz!Box config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+
+    def __init__(self):
+        """Initialize flow."""
+        self._host = None
+        self._manufacturer = None
+        self._model = None
+        self._name = None
+        self._password = None
+        self._username = None
+
+    def _get_entry(self):
+        return self.async_create_entry(
+            title=self._name,
+            data={
+                CONF_HOST: self._host,
+                CONF_PASSWORD: self._password,
+                CONF_USERNAME: self._username,
+            },
+        )
+
+    def _try_connect(self):
+        """Try to connect and check auth."""
+        fritzbox = Fritzhome(
+            host=self._host, user=self._username, password=self._password
+        )
+        try:
+            fritzbox.login()
+            fritzbox.logout()
+            return RESULT_SUCCESS
+        except OSError:
+            return RESULT_NOT_FOUND
+        except LoginError:
+            return RESULT_AUTH_FAILED
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            ip_address = await self.hass.async_add_executor_job(
+                socket.gethostbyname, user_input[CONF_HOST]
+            )
+
+            await self.async_set_unique_id(ip_address)
+            self._abort_if_unique_id_configured()
+
+            self._host = user_input[CONF_HOST]
+            self._name = user_input[CONF_HOST]
+            self._password = user_input[CONF_PASSWORD]
+            self._username = user_input[CONF_USERNAME]
+
+            result = await self.hass.async_add_executor_job(self._try_connect)
+
+            if result == RESULT_AUTH_FAILED:
+                errors["base"] = result
+            elif result != RESULT_SUCCESS:
+                return self.async_abort(reason=result)
+            else:
+                return self._get_entry()
```

Comment:
Start with this case, then the abort case and end with the case that doesn't return. Then we can remove the `else:` and have that case outdented.